product.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of Tryton.  The COPYRIGHT file at the top level of
+# this repository contains the full copyright notices and license terms.
+from trytond.model import ModelSQL, ModelView, fields
+from trytond.pool import PoolMeta, Pool
+from trytond.pyson import Eval
+from trytond.exceptions import UserError
+from trytond.i18n import gettext
+
+
+class Certification(ModelSQL, ModelView):
+    "Certification"
+    __name__ = 'agronomics.certification'
+
+    number = fields.Char('Number')
+    date = fields.Date('Date')
+
+    def get_rec_name(self, name):
+        return self.number
+
+
+class Container(ModelSQL, ModelView):
+    "Container"
+    __name__ = 'agronomics.container'
+
+    name = fields.Char('Name')
+    capacity = fields.Numeric('Capacity', digits=(16, 2))
+
+
+class Template(metaclass=PoolMeta):
+    __name__ = 'product.template'
+
+    agronomic_type = fields.Selection([
+            (None, ''),
+            ('grape', "Grape"),
+            ('do-wort', "DO Wort"),
+            ('not-do-wort', "Not DO Wort"),
+            ('wine', "Wine"),
+            ('bottled-wine', "Bottled Wine"),
+            ], "Agronomic Type", select=True)
+    container = fields.Many2One('agronomics.container', 'Container',
+        states={
+            'invisible': Eval('agronomic_type') != 'bottled-wine',
+        }, depends=['agronomic_type'])
+    capacity = fields.Function(fields.Numeric('Capacity', digits=(16, 2),
+        states={
+            'invisible': Eval('agronomic_type') != 'bottled-wine',
+        }, depends=['agronomic_type']), 'get_capacity',
+        searcher='search_capacity')
+
+    def get_capacity(self, name):
+        if self.container:
+            return self.container.capacity
+
+    @classmethod
+    def search_capacity(cls, name, clause):
+        return [('container.capacity',) + tuple(clause[1:])]
+
+
+class Product(metaclass=PoolMeta):
+    __name__ = 'product.product'
+
+    vintage = fields.Many2Many('product.product-agronomics.crop', 'product',
+        'crop', 'Vintage')
+    variety = fields.Many2Many('product.product-product.taxon', 'product',
+        'variety', 'Variety')
+    do = fields.Many2Many('product.product-agronomics.denomination_of_origin',
```

Comment:
Should probably be "dos" or more possibly even better "origins" or "denominations_of_origin"